src/scripts/quests/QuestLineHelper.ts

```diff
@@ -721,6 +721,83 @@ class QuestLineHelper {
 
         App.game.quests.questLines().push(regiTrioQuestLine);
     }
+
+    //Jirachi Quest
+    public static createJirachiQuestLine() {
+        const jirachiQuestLine = new QuestLine('Wish Maker', 'Harness the power of the Millennium Comet and make a wish!', new GymBadgeRequirement(BadgeEnums.Elite_HoennChampion), GameConstants.BulletinBoards.Hoenn);
+
+        const millenniumFest = new TalkToNPCQuest(MillenniumFest, 'Attend the opening ceremony of the Millennium Festival near Lavaridge Town.');
+        jirachiQuestLine.addQuest(millenniumFest);
+
+        const clownRocket = new CustomQuest(1, 0, 'Defeat Team Rocket Jessie & James at the Millennium Festival near Lavaridge Town.', () => App.game.statistics.temporaryBattleDefeated[GameConstants.getTemporaryBattlesIndex('Clown Jessie & James')]());
+        jirachiQuestLine.addQuest(clownRocket);
+
+        const talkToButler1 = new TalkToNPCQuest(Butler1, 'Learn the legend of the Millennium Comet from Butler near Lavaridge Town.');
+        jirachiQuestLine.addQuest(talkToButler1);
+
+        const clearMtChimney2 = new CustomQuest(1, 0, 'Climb to the Mt. Chimney Crater to get a better view of the Millennium Comet as it passes.', () => App.game.statistics.dungeonsCleared[GameConstants.getDungeonIndex('Mt. Chimney Crater')]());
+        jirachiQuestLine.addQuest(clearMtChimney2);
+
+        const catchAbsol = new CaptureSpecificPokemonQuest('Absol', 'You are being stalked by Absol, the Disaster Pokémon. Capture it.', 1, true);
+        jirachiQuestLine.addQuest(catchAbsol);
+
+        const coccoonHatch = new TalkToNPCQuest(CoccoonHatch, 'Examine the crystalline coccoon Butler gave you while at the Mt. Chimney Crater.');
```

Comment:
this way of writing cocoon comes from games/movie?

Reply:
Officially it is in fact “Cocoon”, from checking the poster art on [bulbapedia](https://m.archives.bulbagarden.net/media/upload/e/ed/M17_English_poster.png)